src/application-insights/azext_applicationinsights/_params.py

```diff
@@ -0,0 +1,44 @@
+        c.argument('resource_group_name')
+
+    with self.argument_context('monitor app-insights events list') as c:
+        c.argument('application', options_list=['--app, -a'], id_part='name', help='GUID, app name, or fully-qualified Azure resource name of Application Insights component. The application GUID may be acquired from the API Access menu item on any Application Insights resource in the Azure portal. If using an application name, please specify resource group.')
+        c.argument('event_type', options_list=['--type'], help='The type of events to retrieve.')
+        c.argument('timespan', options_list=['--timespan', '-t'], help='The timespan over which to retrieve metric values. This is an ISO8601 time period value. If timespan is omitted, a default time range of `PT12H` ("last 12 hours") is used. The actual timespan that is queried may be adjusted by the server based. In all cases, the actual time span used for the query is included in the response.')
+        c.argument('resource_group_name')
+
+    with self.argument_context('monitor app-insights events show') as c:
+        c.argument('application', options_list=['--app, -a'], id_part='name', help='GUID, app name, or fully-qualified Azure resource name of Application Insights component. The application GUID may be acquired from the API Access menu item on any Application Insights resource in the Azure portal. If using an application name, please specify resource group.')
+        c.argument('event_type', options_list=['--type'], help='The type of events to retrieve.')
+        c.argument('event', options_list=['--event'], help='GUID of the event to retrieve. This could be obtained by first listing and filtering events, then selecting an event of interest.')
+        c.argument('timespan', options_list=['--timespan', '-t'], help='The timespan over which to retrieve metric values. This is an ISO8601 time period value. If timespan is omitted, a default time range of `PT12H` ("last 12 hours") is used. The actual timespan that is queried may be adjusted by the server based. In all cases, the actual time span used for the query is included in the response.')
+        c.argument('resource_group_name')
+
+    with self.argument_context('monitor app-insights query') as c:
+        c.argument('application', validator=validate_applications, options_list=['--app', '--apps', '-a'], nargs='+', id_part='name', help='GUID, app name, or fully-qualified Azure resource name of Application Insights component. The application GUID may be acquired from the API Access menu item on any Application Insights resource in the Azure portal. If using an application name, please specify resource group.')
```

Comment:
You can safely ignore the singular application in favor of `apps`. This will make it clear to users that this accepts multiple arguments, whereas the other commands only accept one.

Reply:
@harshdeepgupta removing "--app" as a recognized flag

Reply:
per another comment, I moved the definition of application parameter to top level and only redefine it on this command to include`--apps`.